pyaedt/modeler/cad/component_array.py

```diff
@@ -0,0 +1,843 @@
+    def b_vector_name(self):
+        """B vector name.
+
+        Returns
+        -------
+        str
+           Lattice vector name.
+        """
+        return self._oarray.GetPropValue("B Vector")
+
+    @b_vector_name.setter
+    def b_vector_name(self, val):
+        if val in self.b_vector_choices:
+            self._oarray.SetPropValue("B Vector", val)
+        else:
+            self._logger.warning("B vector name not available")
+
+    @property
+    def a_size(self):
+        """A cell count.
+
+        Returns
+        -------
+        int
+           Number of cells in A direction.
+        """
+        return int(self._app.get_oo_property_value(self._omodel, self.name, "A Cell Count"))
+
+    @a_size.setter
+    def a_size(self, val):  # pragma: no cover
+        # Bug in 2024.1, not possible to change cell count.
+        # self._oarray.SetPropValue("A Cell Count", val)
+        pass
+
+    @property
+    def b_size(self):
+        """B cell count.
+
+        Returns
+        -------
+        int
+           Number of cells in B direction.
+        """
+        return int(self._app.get_oo_property_value(self._omodel, self.name, "B Cell Count"))
+
+    @b_size.setter
+    def b_size(self, val):  # pragma: no cover
+        # Bug in 2024.1, not possible to change cell count.
+        # self._oarray.SetPropValue("B Cell Count", val)
+        pass
+
+    @property
+    def padding_cells(self):
+        """Number of padding cells.
+
+        Returns
+        -------
+        int
+           Number of padding cells.
+        """
+        return int(self._app.get_oo_property_value(self._omodel, self.name, "Padding"))
+
+    @padding_cells.setter
+    def padding_cells(self, val):
+        self._oarray.SetPropValue("Padding", val)
+
+    @property
+    def coordinate_system(self):
+        """Coordinate system.
+
+        Returns
+        -------
+        str
+           Coordinate system name.
+        """
+        cs_dict = self._get_coordinate_system_id()
+        if self._cs_id not in cs_dict.values():
+            self._logger.warning("Coordinate system is not loaded, please save the project.")
+            return "Global"
+        else:
+            return [cs for cs in cs_dict if cs_dict[cs] == self._cs_id][0]
+
+    @coordinate_system.setter
+    def coordinate_system(self, name):
+        cs_dict = self._get_coordinate_system_id()
+        if name not in cs_dict.keys():
+            self._logger.warning("Coordinate system is not loaded, please save the project.")
+        else:
+            self._cs_id = cs_dict[name]
+            self._edit_array()
+
+    @property
+    def _array_props(self):
+        """Retrieve the properties of the component array.
+
+        Returns
+        -------
+        dict
+           An ordered dictionary of the properties of the component array.
+        """
+        return self.get_array_props()
+
+    @pyaedt_function_handler()
+    def delete(self):
+        """Delete the array.
```

Comment:
```suggestion
        """Delete the component array.
```
Add "component" for consistency?